java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt

```diff
@@ -595,9 +595,14 @@ fun KotlinFileExtractor.drillIntoParenthesizedExpression(expr: KtExpression): Pa
 
 context(KaSession)
 private fun KotlinFileExtractor.extractExpression(
-    e: KtExpression,
+    e: KtExpression?,
     parent: StmtExprParent
 ): Label<out DbExpr>? {
+    if (e == null) {
+        parent.makeError()
+        return null // TODO: Should return the ID of the error
+    }
```

Comment:
Are we okay with a `tw.getWholeFileLocation()` location for the error expression? We can get a more precise location in QL by returning the location of the parent element. We could also have the same location here if we maintained the stack of psi elements that we process or if we passed the parent too or if `StmtExprParent` had a reference to the parent psi item.

Reply:
I agree we should do something; for now I've added a TODO comment.